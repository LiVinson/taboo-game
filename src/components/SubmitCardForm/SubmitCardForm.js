import React from 'react'
import PropTypes from 'prop-types'
import { Formik, Field, ErrorMessage } from 'formik'
import { Form, FormSection, FormSectionTitle, TextInput, TextLabel, ErrorText } from 'components/shared/FormElements'

//Validation occurs onChange or onBlur of inputs and onSubmit of form
const validate = (values) => {
	const errors = {}
	if (!values.tabooWord) {
		errors.tabooWord = 'A taboo word is required'
	}
	return errors
}

const SubmitCardForm = (props) => {
	return (
		<Formik
			enableReinitialize={true} //allows form to reset if initial state values change
			initialValues={props.initialValues}
			validate={validate}
			onSubmit={(values, actions) => props.handleSubmit(values, actions)}
		>
			{(formik) => (
				<Form id="submitCardForm">
					<FormSection>
						<TextLabel htmlFor="tabooWord">Taboo Word</TextLabel>
						<Field
							id="tabooWord"
							maxLength={35}
							name="tabooWord"
							type="text"
							as={TextInput}
							placeholder=""
							tabIndex="1"
						/>
						<ErrorMessage name="tabooWord" component={ErrorText} />
					</FormSection>

					<FormSection padding="1rem">
						<FormSectionTitle>Words You Can't Say (Optional)</FormSectionTitle>

						<TextLabel htmlFor="word1">Word 1</TextLabel>
						<Field
							id="word1"
							maxLength={35}
							name="word1"
							type="text"
							as={TextInput}
							placeholder=""
							tabIndex="2"
						/>

						<TextLabel htmlFor="word2">Word 2</TextLabel>
						<Field
							id="word2"
							maxLength={35}
							name="word2"
							type="text"
							as={TextInput}
							placeholder=""
							tabIndex="2"
						/>

						<TextLabel htmlFor="word3">Word 3</TextLabel>
						<Field
							id="word3"
							maxLength={35}
							name="word3"
							type="text"
							as={TextInput}
							placeholder=""
							tabIndex="3"
						/>

						<TextLabel htmlFor="word4">Word 4</TextLabel>
						<Field
							id="word4"
							maxLength={35}
							name="word4"
							type="text"
							as={TextInput}
							placeholder=""
							tabIndex="4"
						/>

						<TextLabel htmlFor="word5">Word 5</TextLabel>
						<Field
							id="word5"
							maxLength={35}
							name="word5"
							type="text"
							as={TextInput}
							placeholder=""
							tabIndex="5"
						/>
					</FormSection>
				</Form>
			)}
		</Formik>
	)
}

SubmitCardForm.propTypes = {
	initialValues: PropTypes.object.isRequired,
	handleSubmit: PropTypes.func.isRequired
}

export default SubmitCardForm
