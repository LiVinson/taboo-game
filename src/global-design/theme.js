const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
