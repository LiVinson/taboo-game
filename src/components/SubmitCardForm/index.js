import SubmitCardForm from "./SubmitCardForm"

export default SubmitCardForm