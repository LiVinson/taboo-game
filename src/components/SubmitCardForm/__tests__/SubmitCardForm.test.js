import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import SubmitCardForm from 'components/SubmitCardForm'

test('SubmitCardForm renders correctly', () => {
	const props = {
		handleSubmit: jest.fn(),
		initialValues: {
			tabooWord: '',
			word1: '',
			word2: '',
			word3: '',
			word4: '',
			word5: '',
			successMsg: '',
		},
	}

	const wrapper = renderer.create(
		<ThemeProvider theme={theme}>
			<SubmitCardForm {...props} />
		</ThemeProvider>
	)
	expect(wrapper).toMatchSnapshot()
})
