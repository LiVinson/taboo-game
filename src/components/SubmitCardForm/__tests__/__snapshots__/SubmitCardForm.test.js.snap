// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitCardForm renders correctly 1`] = `
<form
  action="#"
  className="sc-AxjAm hyBwHU"
  id="submitCardForm"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="sc-AxirZ itgPJw"
  >
    <label
      className="sc-Axmtr sc-AxmLO jMZnwH"
      htmlFor="tabooWord"
    >
      Taboo Word
    </label>
    <input
      className="sc-AxhUy dcifbi"
      id="tabooWord"
      maxLength={35}
      name="tabooWord"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      tabIndex="1"
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-AxirZ gKpSjo"
  >
    <p
      className="sc-AxiKw jjCqLL"
    >
      Words You Can't Say (Optional)
    </p>
    <label
      className="sc-Axmtr sc-AxmLO jMZnwH"
      htmlFor="word1"
    >
      Word 1
    </label>
    <input
      className="sc-AxhUy dcifbi"
      id="word1"
      maxLength={35}
      name="word1"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      tabIndex="2"
      type="text"
      value=""
    />
    <label
      className="sc-Axmtr sc-AxmLO jMZnwH"
      htmlFor="word2"
    >
      Word 2
    </label>
    <input
      className="sc-AxhUy dcifbi"
      id="word2"
      maxLength={35}
      name="word2"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      tabIndex="2"
      type="text"
      value=""
    />
    <label
      className="sc-Axmtr sc-AxmLO jMZnwH"
      htmlFor="word3"
    >
      Word 3
    </label>
    <input
      className="sc-AxhUy dcifbi"
      id="word3"
      maxLength={35}
      name="word3"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      tabIndex="3"
      type="text"
      value=""
    />
    <label
      className="sc-Axmtr sc-AxmLO jMZnwH"
      htmlFor="word4"
    >
      Word 4
    </label>
    <input
      className="sc-AxhUy dcifbi"
      id="word4"
      maxLength={35}
      name="word4"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      tabIndex="4"
      type="text"
      value=""
    />
    <label
      className="sc-Axmtr sc-AxmLO jMZnwH"
      htmlFor="word5"
    >
      Word 5
    </label>
    <input
      className="sc-AxhUy dcifbi"
      id="word5"
      maxLength={35}
      name="word5"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      tabIndex="5"
      type="text"
      value=""
    />
  </div>
</form>
`;
